modules/core/test/test_umat.cpp

```diff
@@ -977,6 +977,72 @@ TEST(UMat, DISABLED_Test_same_behaviour_write_and_read)
     ASSERT_TRUE(exceptionDetected); // data race
 }
 
+#if defined HAVE_OPENCL && 0
+
+#include "opencv2/core/opencl/runtime/opencl_core.hpp"
+
+// pure reproducer of bug in clEnqueueReadBuffer on NVIDIA
+// fails on NVIDIA GPUs (particularly, GTX 650)
+TEST(UMat, BigRead)
```

Comment:
My suggestion is to move this file somewhere else, because there are no UMats things (for example, opencl/test_device.cpp)